### DISCUSSION TITLE
NetBox maintainers meeting minutes, 11 Jan 2024

### DISCUSSION THREAD

**Discussion by jeffgdotorg** [netbox-community/netbox#14777] [Announcements]
*NetBox maintainers meeting minutes, 11 Jan 2024*

_4.0 milestones (Jeremy)_
- Done: Dropped support for Python 3.8 & 3.9, added 3.12
- Done: Replaced Bleach
- WIP: Replace `graphene-django` library with Strawberry (Arthur)
- WIP: Tabler for web UI (Jeremy)

_Downgrading gunicorn? ([NetBox bug](https://github.com/netbox-community/netbox/issues/14691)) (Jeremy)_
- Upstream bug is [benoit/gunicorn#3038](https://github.com/benoitc/gunicorn/issues/3038)
- If we downgrade gunicorn, we risk losing explicit support for Python 3.11
- *Decision: Not downgrading gunicorn.*
- Discussion: If we find that gunicorn is becoming less well maintained, should we consider moving to uWSGI or another alternative?

_Prefix / IP hierarchy stuff (Dan Sheppard)_
- Classifying as a _4.0-maybe_ any work to address prefix / IP hierarchy buglets. We want to keep a tight focus in that release on housekeeping and infrastructure investments. 📌

_Future of reports (Jeremy)_
- Discussion: We’re considering merging reports into scripts and eliminating reports as a separate model entity.
- *Decision:* No opposition -> *tagging this change for 4.0*. There’s work left to do on fleshing out the particulars.
- ↳ Arthur raises: We had previously talked about making scripts a proper entity in the model ([#14438](https://github.com/netbox-community/netbox/issues/14438)). Currently we model report modules and script modules in the DB, but not the reports or scripts themselves. This gets back to the targeting of event rules.
- ↳ Jeremy: Probably worth bringing this into scope for 4.0.
- Related: Pagination for output of reports and scripts is tagged for 4.0.

_Updating translations (Jeremy)_
- Requests for new translations are tracked as GH issues. Proposed fixes to existing translations are not – we’re handling that flow directly in [Transifex](https://explore.transifex.com/netbox-community/netbox/), and there’s now a separate button in the project’s new issue flow that sends you over there. NetBox Labs is covering the cost of machine translation.
- ↳ The one exception to the no-GH-issue-for-translation-fixes policy is if there’s a string that’s not getting translated. This hasn’t happened yet.
- When joining a NetBox translation project on Transifex, there’s an approval step that currently goes through Jeremy. Notifications for these requests seem lacking, so please be patient while we work out the kinks.
- There’s still process work that we need to figure out concerning how translations fit into the release process.
- ↳ Abhi raises: Is there a way to do this for plugins?
- ↳ Discussion: Great question. Unknown for now, this is a good area for interested people to explore.

_Import change to plugins (Arthur)_
- *Heads up:* In 4.0 we have *dropped backwards compatibility* for the `extras.plugins` mechanism. This change breaks pretty much all plugins, so we need a push pre-4.0 to have all plugin authors update their code.

_Door-hanger items_
- There’s something of a back-backlog for 4.0 that is under consideration and may end up being in if we find ourselves with spare capacity.
- The window is closing for *plugin authors* to register opposition to [#12325](https://github.com/netbox-community/netbox/issues/12325) (*removing the Django admin UI*).

**Comment by jcollie** [comment]
Thanks for posting this! As someone that relies heavily on NetBox at work, but doesn't have time to follow developer discussions I appreciate summaries of ongoing development work.
